Modules/_asynciomodule.c

```diff
@@ -2738,7 +2738,11 @@ task_call_step_soon(asyncio_state *state, TaskObj *task, PyObject *arg)
         return -1;
     }
 
+    // Beware: An evil call_soon could alter task_context.
+    // See: https://github.com/python/cpython/issues/126080.
+    PyObject *task_context = Py_NewRef(task->task_context);
     int ret = call_soon(state, task->task_loop, cb, NULL, task->task_context);
+    Py_DECREF(task_context);
```

Comment:
Why not increment the context in `call_soon` to make this a more global change?

Reply:
1) it might not always be needed
2) we transfer ownership when possible to avoid an INCREF/DECREF pair
3) logic should be separate IMO (same reason why we do not do it in rich comparison)

Reply:
`INCREF` and `DECREF` aren't expensive operations, I would rather get rid of future maintenance if we can.

Reply:
Well, it's still not a universal solution because you might have recursive calls and references that could disappear. We had those kind of issues in `OrderedDict.__eq__` and there was already a discussion on doing it automatically: https://github.com/python/cpython/issues/119004#issuecomment-2108428409. 

Similarly, https://github.com/python/cpython/issues/82769#issuecomment-1093845239 did not consider doing it automatically. Finally, Kumar wanted to reduce the number of `Py_INCREF/Py_DECREF` in general (that was the reason why we used ownership transfer instead).

cc @kumaraditya303 What do you think of doing it only for call_soon though?

Reply:
Yeah, reducing the number of reference count operations is good, but I'm worried that we'll actually end up adding more by doing this on a case-by-case basis.

Reply:
Well, the thing is that each call may or may not be subject to a UAF. Some parts of the code are not UAF-vulnerable because the Python code won't be able to mutate the attribute during the call. It's important to check the performances as well in large benchmarks (`__eq__` is used quite often and having `Py_INCREF/Py_DECREF` *by default* may have undesired performance impact; this must be benchmarked properly though).